test/Dialect/FIRRTL/layers.mlir

```diff
@@ -0,0 +1,36 @@
+// RUN: circt-opt %s
+
+firrtl.circuit "Test" {
+  firrtl.module @Test() {}
+
+  firrtl.layer @A bind {}
+
+  firrtl.module @WhenUnderLayer(in %test: !firrtl.uint<1>) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    firrtl.layerblock @A {
+      %w = firrtl.wire : !firrtl.uint<1>
+      firrtl.when %test : !firrtl.uint<1> {
```

Comment:
I don't know what's happening to this test, but:

* Tests that check that valid IR (regression test the verifier w.r.t. what's being fixed here so it stays fixed)..?
* A test for the new error path(s) and resulting diagnostics, please.  Thanks!

Reply:
I had the same question about the error path. However, it's already covered. 😉 The test is just checking that certain circuits which were previously rejected are now accepted.

Reply:
Oh, I missed the "connect" diagnostic was in the previous code, oops.

Hooray already covered, thanks!  And thanks for explanation.

Reply:
> Tests that check that valid IR (regression test the verifier w.r.t. what's being fixed here so it stays fixed)..?

Yeah two are regression tests that used to fail verification (WhenUnderLayer, and DefineIntoOpenBundle), while the third test (ProbeEscapeLayer) is just a sanity check that I didn't break a pattern that was accepted before. There aren't any new error paths, I've just relaxed the existing checks.